tbx/core/blocks.py

```diff
@@ -320,27 +328,59 @@ class Meta:
         template = "patterns/molecules/streamfield/blocks/work_chooser_block.html"
 
 
-class StoryBlock(StreamBlock):
-    h2 = CharBlock(
+class PhotoCollageStructValue(blocks.StructValue):
+    def url(self):
+        return self.get("page").url
+
+    def button_text(self):
```

Comment:
I wouldn't worry about falling back to the page title here. I don't think we'd want the page title appearing in the button as it would be too long - just using the link text is fine. Also, if it is easier don't worry about creating special methods to match the variables I've used, as I can easily update them.

Reply:
I stand by whatever decisions you and Helen make here.